tests/providers/google/cloud/triggers/test_dataproc.py

```diff
@@ -375,3 +388,90 @@ async def test_async_operation_triggers_on_error(self, mock_hook, operation_trig
         )
         actual_event = await operation_trigger.run().asend(None)
         assert expected_event == actual_event
+
+
+@pytest.mark.db_test
+class TestDataprocSubmitTrigger:
+    def test_submit_trigger_serialization(self, submit_trigger):
+        """Test that the trigger serializes its configuration correctly."""
+        classpath, kwargs = submit_trigger.serialize()
+        assert classpath == "airflow.providers.google.cloud.triggers.dataproc.DataprocSubmitTrigger"
+        assert kwargs == {
+            "job_id": TEST_JOB_ID,
+            "project_id": TEST_PROJECT_ID,
+            "region": TEST_REGION,
+            "gcp_conn_id": TEST_GCP_CONN_ID,
+            "polling_interval_seconds": TEST_POLL_INTERVAL,
+            "cancel_on_kill": True,
+            "impersonation_chain": None,
+        }
+
+    @pytest.mark.asyncio
+    @mock.patch("airflow.providers.google.cloud.triggers.dataproc.DataprocSubmitTrigger.get_async_hook")
+    async def test_submit_trigger_run_success(self, mock_get_async_hook, submit_trigger):
+        """Test the trigger correctly handles a job completion."""
+        mock_hook = mock_get_async_hook.return_value
+        mock_hook.get_job = mock.AsyncMock(
+            return_value=mock.AsyncMock(status=mock.AsyncMock(state=JobStatus.State.DONE))
+        )
+
+        async_gen = submit_trigger.run()
+        event = await async_gen.asend(None)
+        expected_event = TriggerEvent(
+            {"job_id": TEST_JOB_ID, "job_state": JobStatus.State.DONE, "job": mock_hook.get_job.return_value}
+        )
+        assert event.payload == expected_event.payload
+
+    @pytest.mark.asyncio
+    @mock.patch("airflow.providers.google.cloud.triggers.dataproc.DataprocSubmitTrigger.get_async_hook")
+    async def test_submit_trigger_run_error(self, mock_get_async_hook, submit_trigger):
+        """Test the trigger correctly handles a job error."""
+        mock_hook = mock_get_async_hook.return_value
+        mock_hook.get_job = mock.AsyncMock(
+            return_value=mock.AsyncMock(status=mock.AsyncMock(state=JobStatus.State.ERROR))
+        )
+
+        async_gen = submit_trigger.run()
+        event = await async_gen.asend(None)
+        expected_event = TriggerEvent(
+            {"job_id": TEST_JOB_ID, "job_state": JobStatus.State.ERROR, "job": mock_hook.get_job.return_value}
+        )
+        assert event.payload == expected_event.payload
+
+    import asyncio
+    from unittest.mock import AsyncMock, patch
+
+    import pytest
```

Comment:
Why do we want to import them here?

Reply:
Thanks @Lee-W I missed them why debugging